File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.go
func characterReplacement(s string, k int) int {
    count := make(map[byte]int)
	res,l,max:= 0,0,0
	for r:=0;r<len(s);r++{
		count[s[r]]++
		if count[s[r]] > max{
			max = count[s[r]]
		}
		for (r-l+1)-max > k {
			count[s[l]]--
			l++
		}
		if (r-l+1)>res{
			res = r-l+1
		}
	}
	return res
}

File: Data Structures & Algorithms/capacity-to-ship-packages-within-d-days/submission-1.py
class Solution:
    def shipWithinDays(self, weights: List[int], days: int) -> int:
        l,r = max(weights), sum(weights)

        res = r

        while l <= r:
            capacity = (l+r)//2

            dayCount = 1
            currLoad = 0

            for wei in weights:
                if currLoad+wei > capacity:
                    dayCount +=1
                    currLoad = 0

                currLoad+=wei

            if dayCount <= days:
                res = capacity
                r = capacity -1

            else:
                l = capacity +1

        return res 
